### PR TITLE
refactor(mobile): backup album selection

### LINKED ISSUE #5302: [BUG]The mobile backup album page keeps loading
### The bug

There are 8000+ photos and 2300+ videos on the phone.
When the mobile backup album page is opened, assign permissions to immich to access all photos. The backup album page will always show Loading,I've been waiting for this page for about 2-3 hours. as shown in the picture below.
微信图片_20231122204956
Did some tests:
1. On another mobile phone, the number of photos is 1000+, and when connected to the same immich server, the backup album page is displayed normally.
2. On another mobile phone, there are 7000+ photos and 2000+ videos, and the backup album page keeps loading.
3. Set up a mobile phone with a large number of photos, only give immich access to some photos, and the backup album page will display normally.
From the client to the server, they are all within the LAN, and there is no proxy in between.

### The OS that Immich Server is running on

Installing through Docker Compose on Mac OS

### Version of Immich Server

1.88.2

### Version of Immich Mobile App

1.88.0

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
no changes
```

### Your .env content

```Shell
no changes
```

### Reproduction steps

```bash
1.Open the backup album page on the mobile terminal
2.
3.
...
```

### Additional information

_No response_

### LINKED ISSUE #5650: [BUG] Backup Albums are not listed and can't be selected
### The bug

I can't select my backup albums anymore.
Since the last update – when the iCloud photo support was introduced – I can't select the backup albums anymore. The following error is visible in the logs. On the 'Select Albums' page, the loading spinner just keeps spinning.

  

### The OS that Immich Server is running on

Raspbian

### Version of Immich Server

v1.88.2

### Version of Immich Mobile App

v1.90.0

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
_I don't have access to that atm. If still relevant, I can post it later._
```

### Your .env content

```Shell
_I don't have access to that atm. If still relevant, I can post it later._
```

### Reproduction steps

```bash
1. Launch App
2. Open the screen to select the backup albums
3. The loading spinner is the only thing visible.
```

### Additional information

I haven't update the server to the latest version yet. However, the release notes since my current version didn't indicate any requirements for the mobile version.

### LINKED ISSUE #5826: [BUG]Selecting Backup Albums keeps loading
Some phones keep loading when you choose to back up your albums.
The images and log files are as follows
IMG_8075
[Immich_log_2023-12-18T213155.266182.csv](https://github.com/immich-app/immich/files/13705311/Immich_log_2023-12-18T21.31.55.266182.csv)

### The OS that Immich Server is running on

Installing through Docker Compose on Mac OS

### Version of Immich Server

1.91.3

### Version of Immich Mobile App

1.91.0

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
no changes
```

### Your .env content

```Shell
no changes
```

### Reproduction steps

```bash
1. I have two ios phones, both have the same app version and connect to the same server. phone A has 1000+ photos and phone B has 8000+ photos. Only phone B will have this problem.
2. The problem occurs when setting the access permission of immich photos to "all photos" on B phone. So I tried to set the access permission of immich photos to "selected photos", then manually select all photos in the album, then select backup album in the app can be displayed normally!
3. I'm still having this problem whether I choose to turn on the ignore icloud photos feature or not
4. After loading the backup albums via 2. I realized that the list of backup albums in immich doesn't match the list of albums in the phone, but matches the list of Media Types in the phone, and is missing the Screen Recordings.
```

### Additional information

_No response_

### LINKED ISSUE #7711: Mobile App does not correctly handle the selected backup folder, if it is/becomes empty.
### The bug

If I empty the selected backup folder on my mobile phone, then Immich App will kick this folder out of the backup folder list.
Like the screenshots show in the below.

Now "Camera" is selected and there are 4 photos in it.
Screenshot_20240307-183341~2
Screenshot_20240307-183351~2

Then I have deleted all the photos in "Camera"
Screenshot_20240307-183437~2

even you can't select a folder as backup folder if it is currently empty.
Screenshot_20240307-183457_Mosaic_19_59_49

### The OS that Immich Server is running on

Docker Desktop on Windows

### Version of Immich Server

v1.97.0

### Version of Immich Mobile App

v1.97.0

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
nothing special, default version with customized port number
```

### Your .env content

```Shell
nothing special, default version with customized path
```

### Reproduction steps

```bash
1.just delete all the photos in the selected backup folder
```

### Additional information

This bug is not just introduced recently, it exists already long time. But I can't remember the start-point.

### PR DESCRIPTION
**Pull request by alextran1502**
This PR addresses the following issues:

- [X] Albums with 0 assets weren't shown, which causes the loss of selection states when the user removes all the assets from the devices
- [X] Remove the album selection thumbnail in exchange with the check and cross icon to avoid issues when reading the album's thumbnail, which can lead to not showing any albums in the album backup selection page
- [x] Handle accessing the backup page when the background task is still running, leading to no information being displayed on the screen.

Fixes #7711
Fixes #5826
Fixes #5650
Fixes #5302

### CODE REVIEW

**Review by fyfrey — APPROVED** [review]
Lovely, so much simpler!

**Review by shenlong-tanwen — APPROVED** [review]

**Comment by dvinihan** [comment]
@alextran1502 I'm on Android app v1.101.0 and still seeing the issue where albums with 0 assets are not showing up. Did this bug get reintroduced somehow?

**Comment by alextran1502** [comment]
@dvinihan i don't believe so, we did change anything regarding this last update

**Comment by dvinihan** [comment]
@alextran1502 That's strange, I'm still seeing the empty album disappear, and it does not reappear when I add a new photo. 

Here's a screenshot of my logs after I emptied the album. 
Screenshot_20240409_152439

**Comment by alextran1502** [comment]
@dvinihan Can you click on the log and copy the stack trace?

**Comment by dvinihan** [comment]
The 'selected album not found" log has no stack trace as far as I can tell. I recreated the flow again and exported the logs, attached here.
[Immich logs 2024-04-09T21 02 21.csv](https://github.com/immich-app/immich/files/14926096/Immich.logs.2024-04-09T21.02.21.csv)

**Comment by dvinihan** [comment]
@alextran1502 I've played around with this a bit and here's the behavior I'm seeing based on different action. 

In each scenario I started with these steps:
1. Begin with empty Camera folder
2. Add photo to Camera folder
3. In Immich Backup menu, select Camera folder to be backed up
4. Touch "Start Backup"; Photo is successfully backed up

Then, afterwards, based on what actions I took, I saw slightly different results:
**Scenario A:**
1. Delete local copy through Immich interface
2. Open backup menu; Camera is still listed in the "Backup Albums" summary on the general Backup menu. Note that I did not navigate further into the "Select" option here where it lets me actually choose albums.
3. Add a new photo to Camera folder
4. Photo appears and is automatically backed up. Success!

**Scenario B:**
1. Delete local copy through Immich interface
2. Open backup menu; Camera is still listed in the "Backup Albums" summary on the general Backup menu.
3.  Navigate further into the "Select" option. The Camera folder is not present, since it's empty. When I back out to the general Backup menu, Camera is no longer listed in the summary, and new photos are not backed up.

**Scenario C:**
1. Delete local copy through device file manager
2. Open backup menu; Camera is **not** listed in the "Backup Albums" summary on the general Backup menu.
3. New photos in Camera are not backed up.

In summary: Scenario A is the only successful one, where I used the Immich interface to delete the local photo and did not navigate into the backup album selector.

**Comment by alextran1502** [comment]
@dvinihan Thank you for the detail write up, this will help us debug this case better
